### PR TITLE
Update pmodel overview with team and pws info

### LINKED ISSUE #430: Update Programming Model Overview
Update programming model overview to include new features in OpenSHMEM 1.5: put-with-sign, signal fetch, and teams. Anything else (skim changelog to check)?

### PR DESCRIPTION
**Pull request by naveen-rn**
Updating the programming model overview section in the introduction
segment with the team and put-with-signal information

Closes #430

### CODE REVIEW

**Comment by naveen-rn** [comment]
@jdinan Initial update to the programming models overview section - please update as needed.

File: content/programming_model_overview.tex [hunk 1]
```diff
@@ -59,6 +59,16 @@
         performed by the application.
 \end{enumerate}
 
+\item \textbf{Team Management}
+\begin{enumerate}
+    \item \OPR{Teams}: Teams are process subsets created by grouping a set of
```
Comment (jdinan) [inline content/programming_model_overview.tex:64]:
```suggestion
    \item \OPR{Teams}: Teams are \ac{PE} subsets created by grouping a set of
```

File: content/programming_model_overview.tex [hunk 2]
```diff
@@ -59,6 +59,16 @@
...
+    \acp{PE}. Teams are involved in both collective and point-to-point
+    communication operations. Collective communication operations are performed
+    on all \acp{PE} in a valid team and point-to-point communication operations
+    are performed between a local and remote \ac{PE} with team-related \ac{PE}
```
Comment (jdinan) [inline content/programming_model_overview.tex:68]:
```suggestion
    are performed between a local and remote \ac{PE} with team-based \ac{PE}
```

File: content/programming_model_overview.tex [hunk 3]
```diff
@@ -96,6 +106,14 @@
       and returns the old value.
 \end{enumerate}
 
+\item \textbf{Signaling Operations}
+\begin{enumerate}
+  \item \OPR{Signaling Put}: The local \ac{PE} specifies the \source{} data
+        object, private or symmetric, that is copied to the symmetric data
+        object on the remote \ac{PE} and it subsequently updates a flag
+        on the remote \ac{PE} to signal completion.
```
Comment (jdinan) [inline content/programming_model_overview.tex:114]:
```suggestion
  \item \OPR{Signaling Put}: The \source{} data is copied to the symmetric
        object on the remote \ac{PE} and a flag on the remote \ac{PE} is subsequently
        updated to signal completion.
```

**Pull request merged**